File: experiments/runs/2026-09-17_00-24-43_claim-office-example-mapping_exact-ptdd-v1.1-refactor-subagent-cc_opus-5-no-thinking/src/mhpco-rounding.ts
/**
 * MHPCO's rounding rule: every final amount is rounded in the office's own
 * favour.
 *
 * This module owns that single ruling. Which way "the office's favour" points
 * depends on which way the money flows — a premium the customer owes is rounded
 * up, an amount the office pays out is rounded down — so the direction is the
 * office's decision, not arithmetic the caller should make for itself.
 *
 * Only final amounts are rounded; intermediate amounts stay fractions.
 */

/** A premium the customer owes MHPCO is rounded up. */
export function roundPremiumInMHPCOFavour(premium: number): number {
  return Math.ceil(premium);
}

/** A payout MHPCO owes the customer is rounded down. */
export function roundPayoutInMHPCOFavour(payout: number): number {
  return Math.floor(payout);
}

File: experiments/runs/2026-09-20_07-18-54_claim-office-java-example-mapping_baseline-inline-tdd-v1-cc_opus-5-no-thinking/src/main/java/MhpcoRounding.java
import java.math.BigDecimal;
import java.math.RoundingMode;

/**
 * Rounding to whole G, always in the MHPCO's favour: premiums it collects are rounded up,
 * payouts it owes are rounded down.
 */
public final class MhpcoRounding {

    private MhpcoRounding() {
    }

    public static int premium(BigDecimal amount) {
        return amount.setScale(0, RoundingMode.CEILING).intValueExact();
    }

    public static int payout(BigDecimal amount) {
        return amount.setScale(0, RoundingMode.FLOOR).intValueExact();
    }
}
